pypddl-parser/pddlparser.py

```diff
@@ -1,421 +1,427 @@
+    p[0] = p[5]
+
+
+def p_require_def(p):
+    '''require_def : LPAREN REQUIREMENTS_KEY require_key_lst RPAREN'''
+    p[0] = p[3]
+
+
+def p_require_key_lst(p):
+    '''require_key_lst : require_key require_key_lst
+                       | require_key'''
+    if len(p) == 2:
+        p[0] = [p[1]]
+    elif len(p) == 3:
+        p[0] = [p[1]] + p[2]
+
+
+def p_require_key(p):
+    '''require_key : STRIPS_KEY
+                   | EQUALITY_KEY
+                   | TYPING_KEY
+                   | PROBABILISTIC_EFFECTS_KEY'''
+    p[0] = str(p[1])
+
+
+def p_types_def(p):
+    '''types_def : LPAREN TYPES_KEY names_lst RPAREN'''
+    p[0] = p[3]
+
+
+def p_predicates_def(p):
+    '''predicates_def : LPAREN PREDICATES_KEY predicate_def_lst RPAREN'''
+    p[0] = p[3]
+
+
+def p_predicate_def_lst(p):
+    '''predicate_def_lst : predicate_def predicate_def_lst
+                         | predicate_def'''
+    if len(p) == 2:
+        p[0] = [p[1]]
+    elif len(p) == 3:
+        p[0] = [p[1]] + p[2]
+
+
+def p_predicate_def(p):
+    '''predicate_def : LPAREN NAME typed_variables_lst RPAREN
+                     | LPAREN NAME RPAREN'''
+    if len(p) == 4:
+        p[0] = Predicate(p[2])
+    elif len(p) == 5:
+        p[0] = Predicate(p[2], p[3])
+
+
+def p_action_def_lst(p):
+    '''action_def_lst : action_def action_def_lst
+                      | action_def'''
+    if len(p) == 2:
+        p[0] = [p[1]]
+    elif len(p) == 3:
+        p[0] = [p[1]] + p[2]
+
+
+def p_action_def(p):
+    '''action_def : LPAREN ACTION_KEY NAME parameters_def action_def_body RPAREN'''
+    p[0] = Action(p[3], p[4], p[5][0], p[5][1])
+
+
+def p_parameters_def(p):
+    '''parameters_def : PARAMETERS_KEY LPAREN typed_variables_lst RPAREN
+                      | PARAMETERS_KEY LPAREN RPAREN'''
+    if len(p) == 4:
+        p[0] = []
+    elif len(p) == 5:
+        p[0] = p[3]
+
+
+def p_action_def_body(p):
+    '''action_def_body : precond_def effects_def'''
+    p[0] = (p[1], p[2])
+
+
+def p_precond_def(p):
+    '''precond_def : PRECONDITION_KEY LPAREN AND_KEY literals_lst RPAREN
+                   | PRECONDITION_KEY literal'''
+    if len(p) == 3:
+        p[0] = [p[2]]
+    elif len(p) == 6:
+        p[0] = p[4]
+
+
+def p_effects_def(p):
+    '''effects_def : EFFECT_KEY LPAREN AND_KEY LPAREN PROBABILISTIC_KEY effects_lst RPAREN RPAREN
+                   | EFFECT_KEY LPAREN AND_KEY effects_lst RPAREN
+                   | EFFECT_KEY effect'''
+    if len(p) == 3:
+        p[0] = [p[2]]
+    elif len(p) == 6:
+        p[0] = p[4]
+    elif len(p) == 9:
+        p[0] = p[6]   
+
+
+def p_effects_lst(p):
+    '''effects_lst : effect effects_lst
+                   | effect'''
+    if len(p) == 2:
+        p[0] = [p[1]]
+    elif len(p) == 3:
+        p[0] = [p[1]] + p[2]
+
+
+def p_effect(p):
+    '''effect : literal
+              | LPAREN PROBABILISTIC_KEY PROBABILITY literal RPAREN

```

Comment:
It would be slightly better to follow the BNF grammar for probabilistic effects...
Please see Section 3.1 and Appendix A.2 of [PPDDL1.0: An Extension to PDDL for Expressing
Planning Domains with Probabilistic Effects](http://reports-archive.adm.cs.cmu.edu/anon/anon/home/ftp/usr0/ftp/2004/CMU-CS-04-167.pdf) ...